libraries/AP_ToshibaCAN/AP_ToshibaCAN.h

```diff
@@ -48,8 +49,22 @@ class AP_ToshibaCAN : public AP_CANDriver {
     // return a bitmask of escs that are "present" which means they are responding to requests.  Bitmask matches RC outputs
     uint16_t get_present_mask() const { return _esc_present_bitmask; }
 
+    uint8_t get_num_motors(void) const override;
+
     // return total usage time in seconds
-    uint32_t get_usage_seconds(uint8_t esc_id) const;
+    bool get_usage_seconds(uint8_t esc_id, uint32_t& usage_sec) const override;
+
+    // get an individual ESC's temperature in degrees if available, returns true on success
+    bool get_temperature(uint8_t esc_id, int16_t& temp) const override;
+
+    // get an individual ESC's rpm if available, returns true on success
+    bool get_rpm(uint8_t esc_id, uint16_t& rpm) const override;
+
+    // get an individual ESC's current in centi-amps if available, returns true on success
+    bool get_current_ca(uint8_t esc_id, uint16_t& amps_ca) const override;
+
+    // return true if we have received any telemetry data
+    uint32_t have_telem_data(uint8_t esc_id) const override { return _telemetry[esc_id].last_update_ms; }
```

Comment:
Move this one to the .cpp file and add:
if (esc_id >= TOSHIBACAN_MAX_NUM_ESCS)

Reply:
Fixed